tap_pardot/client.py

```diff
@@ -1,140 +1,157 @@
         content = response.json()
         error_message = content.get("err")
 
         if error_message:
-            error_code = content["@attributes"]["err_code"] # Error code of 1 is an expired api_key or user_key
+            error_code = content["@attributes"][
+                "err_code"
+            ]  # Error code of 1 is an expired api_key or user_key
 
             if error_code == 1:
                 LOGGER.info("API key or user key expired -- Reauthenticating once")
                 self.login()
-                response = requests.get(url, headers=headers, params=params)
+                response = requests.request(
+                    method, url, headers=self._get_auth_header(), params=params
+                )
                 content = response.json()
 
         return content
-    
-    @backoff.on_exception(backoff.expo,
-                          (PardotException),
-                          giveup=is_not_retryable_pardot_exception,
-                          jitter=None)
-    def describe(self, endpoint, **kwargs):
 
-        describe_url = self.describe_map.get(endpoint)
+    @backoff.on_exception(
+        backoff.expo,
+        (PardotException),
+        giveup=is_not_retryable_pardot_exception,
+        jitter=None,
+    )
+    def describe(self, endpoint, **kwargs):
+        url = (ENDPOINT_BASE + self.describe_url).format(endpoint, self.api_version)
 
-        if describe_url is None:
-            raise Exception("No describe operation for endpoint {}".format(endpoint))
+        params = {"format": "json", "output": "bulk", **kwargs}
 
-        url = (ENDPOINT_BASE + describe_url).format(self.api_version)
+        content = self._make_request("get", url, params)
 
-        headers = self._get_auth_header()
-        params={"format":"json", "output": "bulk", **kwargs}
-        
-        LOGGER.info("%s - Making request to GET endpoint %s, with params %s", endpoint, url, params)
-        content = self._make_request(url, headers, params)
-        
-        error_message = content.get("err")
-        if error_message:
-            error_code = content["@attributes"]["err_code"] # E.g., "15" for login failed
-            raise PardotException("{} - Pardot returned error code {} while describing endpoint. Message: {}".format(endpoint, error_code, error_message), content)
+        self._check_error(content, "describing endpoint")
 
         return content
 
-    @backoff.on_exception(backoff.expo,
-                          (PardotException),
-                          giveup=is_not_retryable_pardot_exception,
-                          jitter=None)
-    def get(self, endpoint, format_params=None, **kwargs):
+    @backoff.on_exception(
+        backoff.expo,
+        (PardotException),
+        giveup=is_not_retryable_pardot_exception,
+        jitter=None,
+    )
+    def _fetch(self, method, endpoint, format_params, **kwargs):
         # Not worrying about a backoff pattern for the spike
```

Comment:
Looks like this snuck in from the spike. Is this comment block needed at all?

Reply:
good call. I moved it to the section where the re-try actually happens and cleaned up a few more comments.